src/cascadia/TerminalConnection/TelnetConnection.cpp

```diff
@@ -0,0 +1,265 @@
+        if (str.size() == 1 && str.at(0) == L'\r')
+        {
+            str = "\r\n";
+        }
+
+        telnetpp::bytes bytes(reinterpret_cast<const uint8_t*>(str.data()), str.size());
+        _session.send(bytes, [=](telnetpp::bytes data) {
+            _socketSend(data);
+        });
+    }
+
+    // Method description:
+    // - ascribes to the ITerminalConnection interface
+    // - resizes the terminal
+    // Arguments:
+    // - the new rows/cols values
+    void TelnetConnection::Resize(uint32_t /*rows*/, uint32_t /*columns*/)
+    {
+        if (!_isConnected())
+        {
+            /*_initialRows = rows;
+            _initialCols = columns;*/
+        }
+        else
+        {
+        }
+    }
+
+    // Method description:
+    // - ascribes to the ITerminalConnection interface
+    // - closes the websocket connection and the output thread
+    void TelnetConnection::Close()
+    {
+        try
+        {
+            if (_transitionToState(ConnectionState::Closing))
+            {
+                _socket.Close();
+                if (_hOutputThread)
+                {
+                    // Tear down our output thread
+                    WaitForSingleObject(_hOutputThread.get(), INFINITE);
+                    _hOutputThread.reset();
+                }
+
+                _transitionToState(ConnectionState::Closed);
+            }
+        }
+        catch (...)
+        {
+            LOG_CAUGHT_EXCEPTION();
+            _transitionToState(ConnectionState::Failed);
+        }
+    }
+
+    // Method description:
+    // - this method bridges the thread to the Azure connection instance
+    // Arguments:
+    // - lpParameter: the Azure connection parameter
+    // Return value:
+    // - the exit code of the thread
+    DWORD WINAPI TelnetConnection::StaticOutputThreadProc(LPVOID lpParameter)
+    {
+        auto pInstance = reinterpret_cast<TelnetConnection*>(lpParameter);
+        return pInstance->_outputThread();
+    }
+
+    // Method description:
+    // - this is the output thread, where we initiate the connection to Azure
+    //  and establish a websocket connection
+    // Return value:
+    // - return status
+    DWORD TelnetConnection::_outputThread()
+    {
+        try
+        {
+            while (true)
+            {
+                if (_isStateOneOf(ConnectionState::Failed))
+                {
+                    _TerminalOutputHandlers(RS_(L"TelnetInternetOrServerIssue") + L"\r\n");
+                    return E_FAIL;
+                }
+                else if (_isStateAtOrBeyond(ConnectionState::Closing))
+                {
+                    return S_FALSE;
+                }
+                else if (_isStateOneOf(ConnectionState::Connecting))
+                {
+                    try
+                    {
+                        const auto host = Windows::Networking::HostName(_hostname);
+                        _socket = Windows::Networking::Sockets::StreamSocket();

```

Comment:
So, `_socket` is already initiailzed because the "zero value" for a C++/WinRT projection is actually a default-constructed runtimeclass. You can probably just call `ConnectAsync` on it.

You _will_ want to null-initialize both `_reader` and `_writer`, however.